File: src/c#/Creational/FactoryMethod/FactoryMethod1/EmployeeFactory.cs
using System;
using Util;

namespace FactoryMethod1
{
    public class EmployeeFactory : IFactory
    {
        public Employee Create()
        {
            Employee employee = new Employee(EmployeeRandomizer.CreateId(), EmployeeRandomizer.CreateName(),
                EmployeeRandomizer.CreateYear(), EmployeeRandomizer.CreateDepartment());
            return employee;
        }
    }
}

File: src/java/org/javaturk/dp/ch03/lsp/rtti/EmployeeFactory.java
package org.javaturk.dp.ch03.lsp.rtti;

import org.javaturk.dp.ch04.factoryMethod.employee.EmployeeRandomizer;

public class EmployeeFactory implements Factory {

	@Override
	public Employee create() {
		Employee employee = null;
		int i = (int) (Math.random() * 3);
		switch (i) {
		case 0:
			employee = new Employee(EmployeeRandomizer.createId(), EmployeeRandomizer.createName(),
					EmployeeRandomizer.createYear(), EmployeeRandomizer.createDepartment());
			break;
		case 1:
			employee = new Manager(EmployeeRandomizer.createId(), EmployeeRandomizer.createName(),
					EmployeeRandomizer.createYear(), EmployeeRandomizer.createDepartment(),
					EmployeeRandomizer.createDepartment());
			break;
		case 2:
			employee = new Director(EmployeeRandomizer.createId(), EmployeeRandomizer.createName(),
					EmployeeRandomizer.createYear(), "Management", "Management", 5000);
			break;
		}
		return employee;
	}
}
